modules/javafx.graphics/src/main/java/com/sun/glass/ui/win/WinTextRangeProvider.java

```diff
@@ -91,6 +91,18 @@ void dispose() {
         peer = 0L;
     }
 
+    private void validateRange() {
+        String text = (String)getAttribute(TEXT);
```

Comment:
It looks like you already have the `text` everywhere you call this. You might consider passing it in as a parameter rather than calling `getAttribute` redundantly.

Reply:
Thanks for the review.
Addressed both the comments. Please take a re-look.

Reply:
Updated in next commit.